unit-tests-jest/numbers.test.js

```diff
@@ -0,0 +1,173 @@
+import { evenNumbers, toDollars, divideBy, multiplyBy } from './numbers';
+
+// Unit tests for evenNumbers
+describe('evenNumbers', () => {
+  it('returns the even numbers of an array', () => {
+    const numbers = [1, 4, 5, 10, 0];
+    const result = evenNumbers(numbers);
+    expect(result).toEqual([4, 10, 0]);
+  });
+});
+
+describe('evenNumbers', () => {
+  it('returns an emtpy array given an empty array', () => {
+    const numbers = [];
+    const result = evenNumbers(numbers);
+    expect(result).toEqual([]);
+  });
+});
+
+describe('evenNumbers', () => {
+  it('returns the even numbers of an array, even if that are strings', () => {
+    const numbers = ['1', '2', 4, '5', 7];
+    const result = evenNumbers(numbers);
+    expect(result).toEqual(['2', 4]);
+  });
+});
+
+describe('evenNumbers', () => {
+  it('returns an error message if not provided an array', () => {
+    const numbers = 'A String';
+    const result = evenNumbers(numbers);
+    expect(result).toEqual('Please provide an array.');
+  });
+});
```

Comment:
I like these edge case tests. With `describe`, we usually put all of them together that test the same function.

It's like "describe evenNumbers -- it returns the even numbers of an array; it returns an error message if not an array" and so on.

The code looks like this:
```
describe('evenNumbers', () => {
  it('returns the even numbers of an array', () => {...});
  it('returns an error message if not provided an array', () => {...});
  ...
});
```